python/src/main/python/drivers/graphicsfuzz_piglit_converter.py

```diff
@@ -0,0 +1,191 @@
+        test_header += str(uniform_args[0:]).strip('[]').replace(',', '') + '\n'
+    test_header += DRAW_COMMAND
+    return test_header
+
+# Returns a list of all uniform declarations in a shader.
+def get_uniforms_from_shader(shader_lines: List) -> List:
+    return (line for line in shader_lines if is_uniform(line))
+
+# Helper function to see if a given string is a GLSL uniform declaration.
+def is_uniform(line: str) -> bool:
+    return ('uniform' in line)
+
+# Returns the first occurrence of a GLSL '#version' string in a list of lines of shader code.
+def get_version_header_from_shader(shader_lines: List) -> str:
+    return next(line for line in shader_lines if is_version_header(line))
+    
+# Helper function to see if a given string is a GLSL preprocessor version string.
+def is_version_header(line: str) -> bool:
+    return (SHADER_VERSION_FLAG in line)
+    
+# Helper function to parse a shader job JSON file into a list of properties.
+# Throws IOError if the file can't be parsed.
+def get_json_properties(shader_job: str) -> List:
+    with open(shader_job, 'r', encoding = 'utf-8', errors = 'ignore') as job:
+        json_parsed = json.load(job)
+    if not json_parsed:
+        raise IOError('Malformed shader job file.')
+    return json_parsed
+
+# Checks to see if a shader job JSON and a corresponding shader file exist at
+# the same location as the shader job. Throws FileNotFoundError if either cannot be found.
+def check_shader_job_exists(shader_job_name: str) -> None:
+    if not os.path.isfile(shader_job_name):
+        raise FileNotFoundError('Input shader job ' + shader_job_name + ' not found.')
+    # We need to make sure a corresponding shader exists as well.
+    shader_name = get_shader_from_job(shader_job_name)
+    # TODO: Figure out how piglit handles compute shaders so we can support them here.
+    if not os.path.isfile(shader_name):
+        raise FileNotFoundError('Fragment shader ' + shader_name + ' not found.')
+
+# Helper function to get the filename of a shader file from its corresponding shader job.
+def get_shader_from_job(shader_job: str) -> str:
+    return shader_job.split(".")[0] + ".frag"
+
+# Helper function to get the filename of the new shader test from the given shader job.
+def get_shader_test_from_job(shader_job: str) -> str:
+    return shader_job.split(".")[0] + ".shader_test"
+
+def main_helper(args: List[str]) -> int:
+    description = (
+        'Given a GraphicsFuzz shader job JSON file, produce a Mesa piglit \
+        shader_test file. The shader test will be the same name as the shader job.')
+
+    argparser = argparse.ArgumentParser(description = description)
+
+    argparser.add_argument(
+        'shader_job', 
+        help = ('Path to the GraphicsFuzz shader job JSON file.'))
+    
+    args = argparser.parse_args(args)
+
+    # Make sure the shader job exists and that the given path to write to is valid.
+    check_shader_job_exists(args.shader_job)
```

Comment:
Nit: it is common in our code base to do a *basic* check of all arguments, so that simple errors are caught early. So this is good in that it follows our style. However, in this case, I would suggest removing this, as (a) the check is somewhat non-trivial and (b) the first thing `make_shader_test_string` does is read the JSON and then read the frag file; if these fail, exceptions will be raised. The Python approach is to just do things and see if an exception is raised, instead of `if check: do thing`. We don't always follow this though, and this is a subjective!

Reply:
Fair enough - when I was originally writing this I planned to let the user choose their own directory to write to as a second argument, but when I dropped that I still left this check in.